### PR TITLE
test/call: Add test_call_change_videodir

### PR DESCRIPTION
**Pull request by maximilianfridrich**
In response to #2073.

### CODE REVIEW

File: test/call.c [hunk 1]
```diff
@@ -884,6 +898,104 @@ int test_call_video(void)
...
+	ASSERT_TRUE(a_video_rdir == SDP_SENDRECV);
+	ASSERT_TRUE(b_video_ldir == SDP_SENDRECV);
+	ASSERT_TRUE(b_video_rdir == SDP_INACTIVE);
+
+	/* Set video sendrecv */
+	err = call_set_video_dir(ua_call(f->a.ua), SDP_SENDRECV);
+	TEST_ERR(err);
+	err = re_main_timeout(10000);
+	err |= re_main_timeout(10000);
+	err |= re_main_timeout(10000);
```
Comment (cspiel1) [inline test/call.c:969]: The `if` in the `event_handler()` should be extended that these multiple `re_main_timeout()` become one.

File: test/call.c [hunk 2]
```diff
@@ -365,6 +366,17 @@ static void event_handler(struct ua *ua, enum ua_event ev,
...
 		break;
 
+	case UA_EVENT_CALL_REMOTE_SDP:
+		if (f->accept_session_updates &&
+		    call_state(call) == CALL_STATE_ESTABLISHED &&
+		    !call_is_outgoing(call) &&
+		    sdp_media_ldir(stream_sdpmedia(video_strm(call_video(
+				call)))) == SDP_SENDRECV &&
+		    sdp_media_rdir(stream_sdpmedia(video_strm(call_video(
+				call)))) == SDP_INACTIVE) {
```
Comment (cspiel1) [inline test/call.c:376]: Maybe we could use a "state" variable set in the test-routine instead of the long if condition.

File: test/call.c [hunk 3]
```diff
@@ -412,6 +424,20 @@ static void event_handler(struct ua *ua, enum ua_event ev,
...
+		    !call_is_outgoing(call) &&
+		    0 == str_casecmp(prm, "video")&&
+		    sdp_media_ldir(stream_sdpmedia(video_strm(call_video(
+				ua_call(f->a.ua))))) == SDP_SENDRECV &&
+		    sdp_media_rdir(stream_sdpmedia(video_strm(call_video(
+				ua_call(f->a.ua))))) == SDP_SENDRECV &&
+		    sdp_media_ldir(stream_sdpmedia(video_strm(call_video(
+				ua_call(f->b.ua))))) == SDP_SENDRECV &&
+		    sdp_media_rdir(stream_sdpmedia(video_strm(call_video(
+				ua_call(f->b.ua))))) == SDP_SENDRECV) {
```
Comment (cspiel1) [inline test/call.c:438]: Same here. A state variable.

File: test/call.c [hunk 4]
```diff
@@ -884,6 +912,102 @@ int test_call_video(void)
...
+	TEST_ERR(fix.err);
+
+	/* verify that video was enabled for this call */
+	ASSERT_EQ(1, fix.a.n_established);
+	ASSERT_EQ(1, fix.b.n_established);
+
+	ASSERT_TRUE(call_has_video(ua_call(f->a.ua)));
+	ASSERT_TRUE(call_has_video(ua_call(f->b.ua)));
+
+	/* Set video inactive */
```
Comment (cspiel1) [inline test/call.c:956]: Set "state" here to "inactive requested".

File: test/call.c [hunk 5]
```diff
@@ -884,6 +912,102 @@ int test_call_video(void)
...
+			video_strm(call_video(ua_call(f->b.ua)))));
+	b_video_rdir = sdp_media_rdir(stream_sdpmedia(
+			video_strm(call_video(ua_call(f->b.ua)))));
+
+	ASSERT_TRUE(a_video_ldir == SDP_INACTIVE);
+	ASSERT_TRUE(a_video_rdir == SDP_SENDRECV);
+	ASSERT_TRUE(b_video_ldir == SDP_SENDRECV);
+	ASSERT_TRUE(b_video_rdir == SDP_INACTIVE);
+
+	/* Set video sendrecv */
```
Comment (cspiel1) [inline test/call.c:978]: Set "state" here to "sendrecv requested"!

File: test/call.c (hunk 2 above)
Comment (maximilianfridrich) [reply test/call.c:376]: I tried and couldn't get it to work. These events are triggered too often and not when I would expect it.

File: test/call.c (hunk 3 above)
Comment (maximilianfridrich) [reply test/call.c:438]: I tried and couldn't get it to work. These events are triggered too often and not when I would expect it.

File: test/call.c (hunk 4 above)
Comment (maximilianfridrich) [reply test/call.c:965]: See other comment regarding state.

File: test/call.c (hunk 5 above)
Comment (maximilianfridrich) [reply test/call.c:987]: See other comment regarding state.

**Pull request merged**